### ISSUE TITLE
Create github release

### ISSUE THREAD

**Issue by jekuno** [svenfuchs/gem-release#62]
It would be nice if `gem-release` would offer an option to also create an according new github release when using the `gem release` command (e.g. via `gem release --github`).

Github offers a simple API for this purpose. You can automatically create a new Github release using a simple POST, see https://developer.github.com/v3/repos/releases/#create-a-release.

**Comment by johnnyshields** [comment]
:+1:  this would be useful

**Comment by svenfuchs** [comment]
Created a draft here: https://github.com/svenfuchs/gem-release/pull/73

This is untested (and I do not work with GitHub releases much). 

Would anyone care to test this out?

**Comment by PikachuEXE** [comment]
@jekuno @johnnyshields 
Rebased #73 & fixed heredoc for ruby 2.2
Feel free to test it when available

**Comment by johnnyshields** [comment]
Will try it next time I'm releasing a gem. thanks

**Comment by piotrb** [comment]
Just stumbled into this, this is great, my only other request would be to allow the github token to be retrieved from the environment somehow .. 

Since I use this for publishing to the github package repository, re-using that mechanism would be great .. so perhaps support getting the token from rubygems credentials the same way the `key` option does .. or just use `ENV['GITHUB_TOKEN']` or some such ..

**Comment by PikachuEXE** [comment]
@piotrb 
> support getting the token from rubygems credentials the same way the `key` option does
Can you explain a bit more about this?
Do you mean the `--key` option? (Use the API key from ~/.gem/credentials)

**Comment by piotrb** [comment]
yes I do .. the --key option .. at least for github uses personal access tokens .. which is the same auth you would want to use to get a github token from .. 

Like I said there should probably be 3 options .. (in order of precedence)

1. GITHUB_TOKEN env var
2. detect that the push destination is a github package host, if so, get the token from whatever `--key` is pointing to
3. take the key directly with `--token`

**Comment by PikachuEXE** [comment]
I am fine with `GITHUB_TOKEN` as it seems to be [used by GitHub CLI](https://github.com/cli/cli/blob/v0.11.1/pkg/cmd/api/api.go#L114) too 

But `--key` means `Use the API key from ~/.gem/credentials`
I think that's not where users are expected to store their GitHub access tokens

For first option we might want to support custom env var name as I found 
[an issue in GitHub CLI](https://github.com/cli/cli/issues/1229) about it
